cpp/src/parquet/encoding.cc

```diff
@@ -3369,6 +3369,9 @@ class DeltaByteArrayDecoderImpl : public DecoderImpl, virtual public TypedDecode
       if (ARROW_PREDICT_FALSE(prefix_len_ptr[i] < 0)) {
         throw ParquetException("negative prefix length in DELTA_BYTE_ARRAY");
       }
+      if (prefix_len_ptr[i] == 0) {
```

Comment:
You should probably move this just above the above `if`. Not sure it will make a difference...